cli/cppcheckexecutor.cpp

```diff
@@ -66,11 +66,26 @@
 #include <windows.h>
 #endif
 
+class XMLErrorMessagesLogger : public ErrorLogger
+{
+    void reportOut(const std::string & outmsg, Color /*c*/ = Color::Reset) override
+    {
+        std::cout << outmsg << std::endl;
+    }
+
+    void reportErr(const ErrorMessage &msg) override
+    {
+        reportOut(msg.toXML());
```

Comment:
I have the feeling we want to write errors to std::cerr

Reply:
This is just for printing the XML and moves the previous code which did the same:

```
void CppCheckExecutor::reportErr(const ErrorMessage &msg)
{
    if (mShowAllErrors) {
        reportOut(msg.toXML());
        return;
    }
```

Reply:
is this code only meant to be used if `--errorlist` is used?

Reply:
Yes.
```cpp
if (parser.getShowErrorMessages()) {
            mShowAllErrors = true;
            XMLErrorMessagesLogger xmlLogger;
            std::cout << ErrorMessage::getXMLHeader(settings.cppcheckCfgProductName);
            CppCheck::getErrorMessages(*this);
            CppCheck::getErrorMessages(xmlLogger);
            std::cout << ErrorMessage::getXMLFooter() << std::endl;
        }
```

It should probably be moved somewhere else. I have a refactoring of the `CmdLineParser` which moves some things around which might help with this.

This PR is just about cleaning up the `ErrorLogger` implementation in `CppcheckExecutor` a bit.